src/fprime_openmct/fprime_telem_poller.py

```diff
@@ -0,0 +1,137 @@
+
+            #check for structs
+            if isinstance(hist.get_val(), dict):
+                #print(hist.get_val().items())
+                i = 0
+                for key, val in hist.get_val().items():
+                    hist_name = str(hist.template.comp_name) + '.' + str(hist.template.name) + '.' + hist.template.ch_type_obj.__name__[hist.template.ch_type_obj.__name__.find('::'):].replace('::', '') + '.' + key
+                    #print(hist_name)
+
+                    hist_data = {
+                                'name': hist_name, 
+                                'data': {
+                                        'id':hist.id+i,
+                                        'val': val,
+                                        'time':hist.time.to_readable()
+                                        }
+                                }
+                    
+                    self.telem_data.append(hist_data)
+
+                    if self.json_writeable:
+                        self.telem_init_states[hist_name] = val
+
+                    i = i + 1
+
+            else: 
+                hist_name = str(hist.template.comp_name) + '.' + str(hist.template.name)
+
+                hist_data = {
+                            'name': hist_name, 
+                            'data': {
+                                    'id':hist.id,
+                                    'val':hist.get_val(),
+                                    'time':hist.time.to_readable()
+                                    }
+                            }
+                
+                self.telem_data.append(hist_data)
+
+                if self.json_writeable:
+                    self.telem_init_states[hist_name] = hist.get_val()
+
+
+
+
+    def write_telem_json(self, fname="openmct/initial_states.json"):
+        self.telem_init_json = json.dumps(self.telem_init_states, indent=4)
+        #print(len(self.telem_init_states))
+        with open(fname, "w") as outfile:
+            outfile.write(self.telem_init_json)
+
+    def post_telem(self, uri="http://127.0.0.1:4052/fprime_telem"):
+        requests.post(uri, json={'name': 'heli', 'telem': self.telem_data}) 
+
+
+#Set up and Process Command Line Arguments
+arguments, _ = ParserBase.parse_args([StandardPipelineParser, OpenMCTTelemetryPollerParser],
+                                             description="OpenMCT Telemetry Polling Parser",
+                                             client=True  # This is a client script, thus client=True must be specified
+                                             )
+
+# instantiate the GDS and connect to the Deployment
+telem_pipeline = TelemPipeline(connection_ip=arguments.tts_addr, 
+                               connection_port=arguments.tts_port,
+                               dict_path=arguments.dictionary,
+                               log_path=arguments.logs)
+
+# Continuously poll for telemetry from the F-Prime GDS Pipeline
```

Comment:
All this stuff at the file level (from line 111 down) should be moved inside a function (e.g. main()), then this function can be set as the entrypoint for your module in the setup.py, see here for example https://github.com/fprime-community/fprime-tools/blob/77aa704873e5ad0da1d66b26f7a917cc7842e97d/setup.py#L115

Reply:
I went ahead and put all of that code inside main(). I also modified my setup.py to have the main function as an entry point for this executable